core/src/main/scala/MetaDataMacro.scala

```diff
@@ -36,9 +36,9 @@ object MetaDataMacro extends MetaDataMacro {
           case opAnnotation if opAnnotation.tree.tpe <:< weakTypeOf[query] =>
             val name = opAnnotation.tree.children.tail.head
             q"OperationType.Query($name)"
-        }
+        }.getOrElse(EmptyTree)
 
-        q"($methodName -> $operationType.map { o => MethodMetaData(o) })"
+        q"($methodName -> Some(MethodMetaData($operationType)))"
```

Comment:
Instead of doing `getOrElse` and then rewrapping, you can do

```scala
$operationType.map(MethodMetaData(_)).orElse(Some(EmptyTree))
```

or

```scala
val operationType = m.annotations.collectFirst {
  // ...
}.orElse(Some(EmptyTree))

q"($methodName -> $operationType.map(MethodMetadata(_)))"
```